src/Workspaces/Remote/Core/SolutionAssetStorage.Scope.cs

```diff
@@ -67,15 +69,14 @@ public void Dispose()
             AssetHint assetHint, HashSet<Checksum> remainingChecksumsToFind, Dictionary<Checksum, object> result, CancellationToken cancellationToken)
         {
             var solutionState = this.CompilationState;
-            if (solutionState.TryGetStateChecksums(out var stateChecksums))
-                await stateChecksums.FindAsync(solutionState, assetHint, remainingChecksumsToFind, result, cancellationToken).ConfigureAwait(false);
-
-            foreach (var projectId in solutionState.SolutionState.ProjectIds)
+            if (ProjectId is null)
             {
-                if (remainingChecksumsToFind.Count == 0)
-                    break;
-
-                if (solutionState.TryGetStateChecksums(projectId, out stateChecksums))
+                if (solutionState.TryGetStateChecksums(out var stateChecksums))
```

Comment:
instead of TryGet just get the checksums.  Or use TryGet, but contract-call that it must succeed.  (Same as below).